app/routes/survey.$surveyId.tsx

```diff
@@ -0,0 +1,58 @@
+import type { LoaderFunctionArgs } from "@remix-run/node";
+import { Link, json } from "@remix-run/react";
+import { useLoaderData } from "@remix-run/react";
+
+import { prismaDatabase } from "~/prismaDatabase";
+
+
+async function getLoaderData(surveyId: string) {
```

Comment:
Be sure to handle the case where `surveyId` is null. You can throw a 400 to indicate a malformatted request.